demo/agw/ScrolledThumbnail.py

```diff
@@ -0,0 +1,640 @@
+#!/usr/bin/env python
```

Comment:
In general, more comments and docstrings in the sample would be good. The demo modules are used for education about the feature so we should try to support that. Comments about what the code is doing and why it is doing so are good, especially for things that may be non-obvious.

Reply:
Yes, yes. I like documentation, tho I am not good at it.

Reply:
I can add some documentation to ShowDir() in ScrolledThumbnail.py.  That's the only code which actually has some functionality.

The majority of the demo code doesn't do anything more complicated than take a value entered by the user and pass it on to either ThumbnailCtrl or ScrolledThumbnail. There's a bunch of boilerplate code to create popups for the thumbnails and log when they are selected.  I'm not sure that comments on these functions would do more than echo the function name.

What do you think needs additional documentation?

Reply:
besides # comments on lines explaining what the next code lines should do,
or docstrings,...
The most important thing would be to grasp the readers interest, by suggesting what you could *actually* do with the widget in the module docstring at the top of the file along with a small sample "smoke test"

Reply:
Ex: Why did YOU go thru `ALL THE EFFORT` to refactor this for example. What is YOUR use case...

Reply:
That is in the comments in the code for agw/scrolledthumbnail.py.  I'm not clear what benefit there is in duplicating this in the demo.

Reply:
Do you really think that adding this would be useful:

    #  Tell widget to highlight thumbnail when selected
    def OnHighlight(self, event):
...
   # Set zoom factor in widget
    def OnSetZoom(self, event):

Reply:
I was meaning more or less what you could do with the widget besides use it as the basic thumbnail ctrl. some examples I could think of would be like a photo collage or photo mosiac type or rotary thumbnail ctrl implementation. stuff along those lines. If users cant get a decent idea of what they want to create then basically most often it is a waste of time for them and they will pass over the widget for another for example.

Reply:
small comments like those are useful also, tho I tend to not like it when folks put them right above the func/meth name. Put that stuff in the docstring. Above the def name is usually reserved for decorators. Usually # comments are inline or right above a line/section of code to better explain maybe some simple mathematics or design idea, that might be a feature/etc.

Reply:
```python
@frozmacate
def Foo():
    """
    Bar baz the frizzly attribute 
    """
    # Add pow to balloon.
    air += 42
    # release flaming flabbergust
    air = air * 0.987654321
    hot(air, baloon)
```

Reply:
There are docstrings in the widget code which describe the functionality of most functions.  There are comments about extending the code. 

This is the demo we are talking about.  Not the actual widget.  

I find that comments that say nothing more than the name of the function to be clutter, not informative.  Almost all of the demo code is exactly the same:  Get a value from the user, send it to the widget.    Many times.  There are no mathematical tricks or design ideas.  Document strings are not going say anything that will make it clearer or more useful.  Adding uninformative docstrings to dozens of functions is makework, not productive.

Reply:
As far as the demo, just try your best. Everyone starts out as a noob and reads that stuff. Try to remember when you was there also and remember some of the things that might have been explained better.

Reply:
> What do you think needs additional documentation?

It doesn't need to be a lot. Just enough for newbies looking at the code for the first time to not be overwhelmed or to get lost.

Reply:
I added documentation to both demos and to the common widget code.